sno/init.py

```diff
@@ -57,6 +57,20 @@ def _add_datasets_to_working_copy(repo, *datasets, replace_existing=False):
     wc.write_full(commit, *datasets)
 
 
+class GenerateIDsFromFile(StringFromFile):
+    name = "ids"
+
+    def convert(self, value, param, ctx):
+        fp = super().convert(
+            value,
+            param,
+            ctx,
+            # Get the file object, so we don't have to read the whole thing
+            as_file=True,
+        )
+        return iter(fp)
```

Comment:
do we need to strip newlines off these? Or do any other validation?

Reply:
apparently yes we need to strip newlines, not sure how I missed that. will do

validation: there's not much I _can_ do easily; any valid string might be a valid PK